plugins/circleci/src/components/Settings/Settings.tsx

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React, { useState } from 'react';
+import {
+  Button,
+  TextField,
+  List,
+  ListItem,
+  Snackbar,
+  Box,
+  Dialog,
+  DialogTitle,
+} from '@material-ui/core';
+import { Alert } from '@material-ui/lab';
+import { useSettings } from '../../state';
+
+const Settings = () => {
+  const [
+    {
+      repo: repoFromStore,
+      owner: ownerFromStore,
+      token: tokenFromStore,
+      showSettings,
+    },
+    { saveSettings, hideSettings },
+  ] = useSettings();
+
+  const [token, setToken] = React.useState(() => tokenFromStore);
```

Comment:
These should just be `useState` and `useEffect`

Reply:
Do you mean useContext should be used directly to pick the settings data from the state in all the other places?

Reply:
I mean you should `import { useState, useEffect }`

Reply:
Got it.